### PR TITLE
PHOENIX-7270 : Always resolve table before DDL operations

### PR DESCRIPTION
**Pull request by palashc**

### CODE REVIEW

File: phoenix-core/src/test/java/org/apache/phoenix/cache/ServerMetadataCacheIT.java [hunk 1]
```diff
@@ -85,13 +89,13 @@
...
 
     private final Random RANDOM = new Random(42);
-    private final long NEVER = (long) ConnectionProperty.UPDATE_CACHE_FREQUENCY.getValue("NEVER");
 
     private static ServerName serverName;
 
     @BeforeClass
     public static synchronized void doSetup() throws Exception {
         Map<String, String> props = Maps.newHashMapWithExpectedSize(1);
+        props.put(QueryServices.DEFAULT_UPDATE_CACHE_FREQUENCY_ATRRIB, Long.toString(Long.MAX_VALUE));
```
Comment (shahrs87) [inline phoenix-core/src/test/java/org/apache/phoenix/cache/ServerMetadataCacheIT.java:98]:
```suggestion
        props.put(QueryServices.DEFAULT_UPDATE_CACHE_FREQUENCY_ATRRIB, (long) ConnectionProperty.UPDATE_CACHE_FREQUENCY.getValue("NEVER"););
```

File: phoenix-core/src/test/java/org/apache/phoenix/cache/ServerMetadataCacheIT.java [hunk 2]
```diff
@@ -1700,9 +1704,109 @@ public void testInheritedIndexOnTenantViews(boolean sameTenantViewNames) throws
         }
     }
 
+    /**
+     * Test that a client can learn about the changes to a base table's
+     * sequence number after a view is created on the table.
+     */
+    @Test
+    public void testBaseTableSequenceNumberAfterCreateView() throws Exception {
```
Comment (shahrs87) [inline phoenix-core/src/test/java/org/apache/phoenix/cache/ServerMetadataCacheIT.java:1712]: Do we need this test? AlterTableIT#testAddingColumnsToTablesAndViews does the exact same thing.

File: phoenix-core-client/src/main/java/org/apache/phoenix/compile/FromCompiler.java [hunk 3]
```diff
@@ -280,7 +280,7 @@ public static ColumnResolver getResolver(NamedTableNode tableNode, PhoenixConnec
 
     public static ColumnResolver getResolver(SingleTableStatement statement, PhoenixConnection connection)
             throws SQLException {
-        SingleTableColumnResolver visitor = new SingleTableColumnResolver(connection, statement.getTable(), true);
```
Comment (shahrs87) [inline phoenix-core-client/src/main/java/org/apache/phoenix/compile/FromCompiler.java:283]: Do we use one of these resolvers when we run `ALTER TABLE <table-name> SET PROPERTY` statement?

Comment (palashc) [reply phoenix-core-client/src/main/java/org/apache/phoenix/compile/FromCompiler.java:283]: Yes. 
Confirmed from SetPropertyIT that setting property on table/view goes through `ExecutableAddColumnStatement` and that uses the resolver change we made.

**Review by shahrs87 — APPROVED** [review]
+1 ltgm.
Will merge once jenkins is green.

**Comment by palashc** [comment]
@shahrs87 Pushed some checkstyle fixes, test results looked good - https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1854/5/testReport/

**Pull request merged**